### PR TITLE
Refactor and improves

### PR DESCRIPTION
**Pull request by miiu96**
- Added index policy
- Refactor in order to index data in multiple clusters

### CODE REVIEW

File: crossIndex/data.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"github.com/ElrondNetwork/elrond-accounts-manager/mappings"
 )
 
+const AccountsPolicyName = "accounts-manager-retention-policy"
```
Comment (bogdan-rosianu) [inline crossIndex/data.go:8]: add comment

File: crossIndex/data.go [hunk 2]
```diff
@@ -47,7 +49,27 @@ var AccountsTemplate = mappings.Object{
...
 	"settings": mappings.Object{
-		"number_of_shards":   1,
-		"number_of_replicas": 1,
+		"number_of_shards":     1,
+		"number_of_replicas":   1,
+		"index.lifecycle.name": AccountsPolicyName,
+	},
+}
+
+var AccountsClonedPolicy = mappings.Object{
```
Comment (bogdan-rosianu) [inline crossIndex/data.go]: add comment

File: elasticClient/client.go [hunk 3]
```diff
@@ -194,6 +199,35 @@ func (ec *esClient) CreateIndexWithMapping(index string, mapping *bytes.Buffer)
 	return nil
 }
 
+func (ec *esClient) PutPolicy(policyName string, policy *bytes.Buffer) error {
```
Comment (bogdan-rosianu) [inline elasticClient/client.go:203]: add comment

File: elasticClient/client.go [hunk 4]
```diff
@@ -194,6 +199,35 @@ func (ec *esClient) CreateIndexWithMapping(index string, mapping *bytes.Buffer)
...
+	res, err := ec.client.ILM.PutLifecycle(
+		policyName,
+		ec.client.ILM.PutLifecycle.WithBody(policy),
+	)
+
+	if err != nil {
```
Comment (bogdan-rosianu) [inline elasticClient/client.go:208]: remove empty line

File: crossIndex/reindexer/reindexer.go [hunk 5]
```diff
@@ -14,36 +14,57 @@ import (
...
 }
 
 var log = logger.GetOrCreate("reindexer")
 
 // New returns a new instance of reindexer
-func New(sourceIndexer crossIndex.ElasticClientHandler, destinationIndexer crossIndex.ElasticClientHandler) (*reindexer, error) {
+func New(
+	sourceIndexer crossIndex.ElasticClientHandler,
+	destinationIndexer []crossIndex.ElasticClientHandler,
+	pathToIndicesConfig string) (*reindexer, error,
```
Comment (bogdan-rosianu) [inline crossIndex/reindexer/reindexer.go:30]: add check for `pathToIndicesConfig`

File: crossIndex/data.go (hunk 1 above)
Comment (miiu96) [reply crossIndex/data.go:8]: added

File: crossIndex/data.go (hunk 2 above)
Comment (miiu96) [reply crossIndex/data.go]: removed.

File: elasticClient/client.go (hunk 3 above)
Comment (miiu96) [reply elasticClient/client.go:203]: added

File: elasticClient/client.go (hunk 4 above)
Comment (miiu96) [reply elasticClient/client.go:208]: done.

File: crossIndex/reindexer/reindexer.go (hunk 5 above)
Comment (miiu96) [reply crossIndex/reindexer/reindexer.go:30]: done.

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**